x/dex/contract/dag.go

```diff
@@ -0,0 +1,77 @@
+package contract
+
+import (
+	"github.com/sei-protocol/sei-chain/utils"
+	"github.com/sei-protocol/sei-chain/x/dex/types"
+)
+
+type node struct {
+	contractAddr  string
+	incomingNodes *utils.StringSet
+}
+
+// Kahn's algorithm
```

Comment:
wonder if this can be generalized for module sorting. Can be addressed in a new pr though

Reply:
yeah that should be possible. A bit sad that golang doesn't support generics so it might get a little verbose.